src/coreclr/jit/lsrabuild.cpp

```diff
@@ -557,6 +557,20 @@ RefPosition* LinearScan::newRefPosition(Interval*    theInterval,
 {
     if (theInterval != nullptr)
     {
+#if defined(TARGET_AMD64)
+        if (mask == RBM_LOWSIMD)
+        {
+            // Constrain if we have to for float/simd types
+            if (varTypeIsFloating(theInterval->registerType) || varTypeIsSIMD(theInterval->registerType))
```

Comment:
This is heavy check to add in `newRefPosition`. Is there a way to move this check (you can have this logic in a helper method or something) to relevant callers which would definitely pass `mask == RBM_LOWSIMD`?

Reply:
At the callsite, since you already have the `GenTree*` for the `RefPosition`, you can do something like this:

```c++
regMaskTP mask = RBM_NONE;
if (varTypeIsFloat(node) || varTypeIsSIMD(node)) {
  mask  = lowSIMDRegs();
}

BuildUse(node, mask);

Reply:
I believe there was a case where specifying the mask at a `BuildUse` callsite did not cover, something related to a contained `GT_INDIR` node whose register set was not the same as its containing node.

Let me see if I can find the example.

Reply:
With this change https://github.com/dotnet/runtime/commit/6cd4d4301957e32af9a318e24428bf33d051bb05 we can see the following example cause some potential issues...

```C#
[MethodImpl(MethodImplOptions.NoInlining)]
static bool Test_Sse41_TestZ_Normal(in Vector128<Int32> x, in Vector128<Int32> y)
{
    return Sse41.TestZ(x, y);
}
```

which lowers to....

```
N001 (  1,  1) [000000] -----------                    t0 =    LCL_VAR   byref  V00 arg0         u:1 (last use) $80
                                                            /--*  t0     byref  
N002 (  7,  5) [000001] ---XG------                    t1 = *  IND       simd16
N003 (  1,  1) [000002] -----------                    t2 =    LCL_VAR   byref  V01 arg1         u:1 (last use) $81
                                                            /--*  t2     byref  
N004 (  7,  5) [000003] -c-XG------                    t3 = *  IND       simd16
                                                            /--*  t1     simd16 
                                                            +--*  t3     simd16 
N005 ( 15, 11) [000004] ---XG------                         *  HWINTRINSIC void   int PTEST <l:$243, c:$242>
```

Since `TestZ` associated lowered instruction `ptest` does not have an EVEX form, we restrict it's operands (which are both float variables that are contained) to the lower SIMD registers. I believe this creates an issue though since if we specify constraints on `t3 = IND` it also propagates to its child node `t2 = LCL_VAR byref ...`. but the nodes have different variable types. Previously, `RBM_NONE` would have been passed down, which would let `newRefPosition` select a set of registers based on ref positions type.

Does this make sense? Essentially, `RBM_LOWSIMD` was a version of `RBM_NONE` to get around this issue.

Reply:
Hhm, I thought you were planning to remove it. Did you forget to push?